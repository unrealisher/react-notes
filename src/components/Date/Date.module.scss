.date {
  font-size: 18px;
  line-height: 22px;
  align-self: flex-end;
}
