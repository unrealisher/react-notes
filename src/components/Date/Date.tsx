import React from "react";
import moment from "moment";
import "moment/locale/ru";

import styles from "./Date.module.scss";

interface IProps {
  created?: number;
}

export const Date = (props: IProps): JSX.Element => {
  const { created } = props;
  moment.locale("ru");
  return <span className={styles.date}>{moment(created).fromNow()}</span>;
};

export default Date;
