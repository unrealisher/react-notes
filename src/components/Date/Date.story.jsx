import React from "react";

import Date from "./Date";

import { storiesOf } from "@storybook/react";

import "./../../index.scss";

storiesOf("Date", module).add("Стандарт", () => <Date created={1551780000000}/>);
