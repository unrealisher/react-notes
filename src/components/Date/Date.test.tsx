import React from "react";
import { shallow } from "enzyme";
import { Date } from "./Date";

describe("Date", () => {
  const created = 100;
  it("correct render", () => {
    const component = shallow(<Date created={created} />);
    expect(component).toMatchSnapshot();
  });
});
