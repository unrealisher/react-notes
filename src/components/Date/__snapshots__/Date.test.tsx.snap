// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Date
    created={100}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "49 лет назад",
      "className": "date",
    },
    "ref": null,
    "rendered": "49 лет назад",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "49 лет назад",
        "className": "date",
      },
      "ref": null,
      "rendered": "49 лет назад",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
